### ISSUE TITLE
Embedded Videos and Image Support

### ISSUE THREAD

**Issue by williamhoos** [sstur/react-rte#294]
Issue #6 was a great issue last year that allows embedding images from a link.

Split into issues if needed:

1. Can we support embedding videos from youtube? Show the main video preview image just like an image and allow playing from the page?

2. Ability to control image size (small/medium/large like gmail or similar?)

**Comment by williamhoos** [comment]
If it is complicated to embed a playable video within the output HTML, perhaps a user friendly way to embed the cover image for the video and that links to play video in the source location in the new or existing tab per #244

**Comment by Aid19801** [comment]
Hi there, was there ever any progress on this?

**Comment by williamhoos** [comment]
we have a fork that needs to be updated with 16.3 but that mostly solves this. We also have an open issue with some tweaks (size of thumbnail and open in lightbox) if anyone wants to polish it up and submit as a PR to the main fork. If we/anyone does it, will project accept it @sstur ? https://github.com/clinwiki-org/clinwiki/issues/218

**Comment by williamhoos** [comment]
@sstur if we do this, will you accept into code base?

**Comment by sstur** [comment]
Yes, I'm totally open to a PR that adds functionality as long as it doesn't break anything for existing users and ideally it should be behind an option to turn it on. 

I've tackled this image problem before and it turned out pretty complex, partly because of some limitations (at the time) of DraftJS, but perhaps newer versions make this easier and more reliable.

**Comment by williamhoos** [comment]
@sstur @iPilotte @machineghost @CmdrDats @githubmaha @andersryanc 

Pinging recent contributors to see if I could pay any of you to do this? We have a more urgent need. Happy to pay someone reasonable amount to do this.

 I think the simplest implementation could be to add the parsing of image url, that when it includes youtube, it inserts the embed video code instead of a normal image src html render. A perhaps more backward compatible and preferred by @sstur would be a separate "video insert" button to go along with image insert button. Eventually this could/should support other video locations like vimeo, but I understand they each need their own code and can be in flux. All I need is youtube support. The only magic is converting the link people put in, extracting the video id and converting it to render the embed link in iframe. http://net-informations.com/q/mis/youtube.html
